### ISSUE TITLE
error: argument --permission-set-arn: expected one argument

### ISSUE THREAD

**Issue by elliott-king** [aws-amplify/amplify-backend#1632]
### Environment information

```plain text
AWS Cloudshell while in IAM Identity center
```

### Description

I am following the [Amplify docs - Configure AWS for local development](https://docs.amplify.aws/react/start/account-setup/#1-create-user-with-amplify-permissions) and have reached the cloudshell command:
```
response=$(aws sso-admin list-instances)
ssoId=$(echo $response | jq '.Instances[0].IdentityStoreId' -r)
ssoArn=$(echo $response | jq '.Instances[0].InstanceArn' -r)
email_json=$(jq -n --arg email "$user_email" '{"Type":"Work","Value":$email}')
response=$(aws identitystore create-user --identity-store-id $ssoId --user-name amplify-admin --display-name 'Amplify Admin' --name Formatted=string,FamilyName=Admin,GivenName=Amplify --emails "$email_json")
userId=$(echo $response | jq '.UserId' -r)
response=$(aws sso-admin create-permission-set --name amplify-policy --instance-arn=$ssoArn --session-duration PT12H)
permissionSetArn=$(echo $response | jq '.PermissionSet.PermissionSetArn' -r)
aws sso-admin attach-managed-policy-to-permission-set --instance-arn $ssoArn --permission-set-arn $permissionSetArn --managed-policy-arn arn:aws:iam::aws:policy/service-role/AmplifyBackendDeployFullAccess
accountId=$(aws sts get-caller-identity | jq '.Account' -r)
aws sso-admin create-account-assignment --instance-arn $ssoArn --target-id $accountId --target-type AWS_ACCOUNT --permission-set-arn $permissionSetArn --principal-type USER --principal-id $userId
# Hit enter
```

The issue is with that last line - I'm getting this output in the cloudshell terminal:
```
[cloudshell-user@ip-<ipaddress> ~]$ aws sso-admin create-account-assignment --instance-arn $ssoArn --target-id $accountId --target-type AWS_ACCOUNT --permission-set-arn $permissionSetArn --principal-type USER --principal-id $userId

usage: aws [options] <command> <subcommand> [<subcommand> ...] [parameters]
To see help text, you can run:

  aws help
  aws <command> help  aws <command> <subcommand> help

aws: error: argument --permission-set-arn: expected one argument
```

This seems to be a dupe of #1569, but that user never resolved it. A maintainer had this question:

> Could you provide us the full output or the error message from the cloudshell?

But I am getting no more output besides the above.

**Comment by edwardfoyle** [comment]
Can you try putting `$permissionSetArn` in double quotes? eg `--permission-set-arn "$permissionSetArn"`

**Comment by elliott-king** [comment]
That kinda helped, but then I started running into other issues.

I deleted my IAM Identity Center configuration and re-created it. I think the first time I chose "enable in only this AWS account" instead of organizations and that created a cascading effect of failures. I _think_ that `$permissionSetArn` was empty from an earlier silent error. 

I think we can chalk this up to user error. It's unclear to me why "enable only in this AWS account" would eventually lead to permissions issues with `aws sso-admin create-permission-set`, but I was being too blasé about it.
